flopy/mbase.py

```diff
@@ -1695,23 +1695,38 @@ def run_model(
         normal_msg[idx] = s.lower()
 
     # Check to make sure that program and namefile exist
-    exe = which(exe_name)
-    if exe is None:
-        if exe_name.lower().endswith(".exe"):
-            # try removing .exe suffix
-            exe = which(exe_name[:-4])
-    if exe is None:
-        # try abspath
-        exe = which(os.path.abspath(exe_name))
-    if exe is None:
-        raise Exception(
-            f"The program {exe_name} does not exist or is not executable."
-        )
+    if os.path.dirname(exe_name) == "":
+        exe = which(exe_name)
+        if exe is None:
+            if exe_name.lower().endswith(".exe"):
+                # try removing .exe suffix
+                exe = which(exe_name[:-4])
+                if exe is not None:
+                    exe_name = exe_name[:-4]
+        if exe is None:
+            raise Exception(
+                f"The program {exe_name} does not exist or is not executable."
+            )
+        else:
+            if not silent:
+                print(
+                    f"FloPy is using the following executable to run the "
+                    f"model: {exe}"
+                )
     else:
-        if not silent:
-            print(
-                f"FloPy is using the following executable to run the model: {exe}"
+        exe_name = os.path.abspath(exe_name)
+        if not os.path.exists(exe_name) and not os.path.exists(
+            f"{exe_name}.exe"
+        ):
+            raise Exception(
+                f"The program {exe_name} does not exist or is not executable."
             )
+        else:
+            if not silent:
+                print(
+                    f"FloPy is using the following executable to run the "
+                    f"model: {exe_name}"
+                )
```

Comment:
It's great to have this change, but I think we want this one as a separate PR.  I understand that that it's easy to sneak it in here, but I think a separate PR is warranted in this case.

Reply:
@langevin-usgs, rolled back exe path feature for another PR